### ISSUE TITLE
Error [TypeError: Cannot read property '1' of null]

### ISSUE THREAD

**Issue by koppert** [matteodelabre/mongoose-beautiful-unique-validation#17]
The error is on the line 91.

valueRegex.exec(rawValues) return null.

I'm using the MongoDB shell version: 3.2.6 and mongoose@4.5.1

I believe the text changed in the newest version of mongodb and regex can not run properly.

{ [MongoError: E11000 duplicate key error collection: pinaopen.players index: name_1 dup key: { : "Bruno Barros" }]
  name: 'MongoError',
  message: 'E11000 duplicate key error collection: pinaopen.players index: name_1 dup key: { : "Bruno Barros" }',
  driver: true,
  code: 11000,
  index: 0,
  errmsg: 'E11000 duplicate key error collection: pinaopen.players index: name_1 dup key: { : "Bruno Barros" }',
  getOperation: [Function],
  toJSON: [Function],
  toString: [Function] }

**Comment by matteodelabre** [comment]
Hi @koppert, using MongoDB v3.2.7 and Mongoose v4.5.1 the tests are passing correctly. Could you provide the piece of code that triggers this error? Thanks for your help. :)

**Comment by nicksellen** [comment]
It's when `rawValues` has a space in it the regex does not match:

`: "ab",` matches
`: "a b",` does not match

The regex in question is `/\s*:\s*(\S*),/g`

My stacktrace is:
```
rawValues=: "Tag label",
TypeError: Cannot read property '1' of null
    at /code/trustroots/node_modules/mongoose-beautiful-unique-validation/index.js:92:55
    at Array.forEach (native)
    at /code/trustroots/node_modules/mongoose-beautiful-unique-validation/index.js:90:19
    at tryCallOne (/code/trustroots/node_modules/promise/lib/core.js:37:12)
    at /code/trustroots/node_modules/promise/lib/core.js:123:15
    at flush (/code/trustroots/node_modules/asap/raw.js:50:29)
    at _combinedTickCallback (internal/process/next_tick.js:67:7)
    at process._tickDomainCallback (internal/process/next_tick.js:122:9)
```

**Comment by nicksellen** [comment]
Added breaking test here #18

**Comment by nicksellen** [comment]
... actually that breaks on a different line: https://travis-ci.org/matteodelabre/mongoose-beautiful-unique-validation/jobs/144058001#L623 .. but I think you get the idea
```
TypeError: Cannot read property '1' of null
    at beautify (/home/travis/build/matteodelabre/mongoose-beautiful-unique-validation/index.js:61:24)
    at /home/travis/build/matteodelabre/mongoose-beautiful-unique-validation/index.js:180:21
    at /home/travis/build/matteodelabre/mongoose-beautiful-unique-validation/node_modules/mongoose/lib/model.js:3331:16
    at /home/travis/build/matteodelabre/mongoose-beautiful-unique-validation/node_modules/mongoose/lib/model.js:188:9
    at /home/travis/build/matteodelabre/mongoose-beautiful-unique-validation/node_modules/kareem/index.js:127:16
    at doNTCallback0 (node.js:430:9)
    at process._tickCallback (node.js:359:13)
```

**Comment by koppert** [comment]
Sorry for the lack of feedback. The problem is in space, really.

**Comment by matteodelabre** [comment]
Should be fixed in version 3.0.1, thanks for your help. :)

**Comment by nicksellen** [comment]
It's fixed when the keys contain spaces, but not for when the values do. There were two causes of the same error message.

**Comment by nicksellen** [comment]
Added breaking test PR #21

**Comment by matteodelabre** [comment]
I overlooked that a bit indeed. Thanks for the heads up!

**Comment by matteodelabre** [comment]
Should be fixed in version 3.0.2. Value parsing is a bit less hacky and uses JSON.parse.

**Comment by nicksellen** [comment]
Super, thanks! Looks good to me here.
